src/Framework/Properties/AssemblyInfo.cs

```diff
@@ -44,6 +44,7 @@
 [assembly: InternalsVisibleTo("MSBuild, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 [assembly: InternalsVisibleTo("Microsoft.Build.Conversion.Core, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 
+[assembly: InternalsVisibleTo("Microsoft.Build.Engine.OM.UnitTests, PublicKey=002400000480000094000000060200000024000052534131000400000100010015c01ae1f50e8cc09ba9eac9147cf8fd9fce2cfe9f8dce4f7301c4132ca9fb50ce8cbf1df4dc18dd4d210e4345c744ecb3365ed327efdbc52603faa5e21daa11234c8c4a73e51f03bf192544581ebe107adee3a34928e39d04e524a9ce729d5090bfd7dad9d10c722c0def9ccc08ff0a03790e48bcd1f9b6c476063e1966a1c4")]
```

Comment:
Should be able to remove this.

Reply:
We use a ChangeWaves class for one of the tests, and it is internal. So, we need it.

Reply:
Are you sure it's still needed? I suggest moving the test from `Microsoft.Build.Engine.OM.UnitTests` to another assembly (`Microsoft.Build.Engine.UnitTests`?) instead of adding this attribute. I believe it's intentional the "OM" test suite accesses only the public Object Model.

Reply:
Sorry, I was wrong. The test is already in `Microsoft.Build.Engine.UnitTests` and is not problematic. The problem is in `ManagedFileSystem` class. It is a shared class and it is used in `Microsoft.Build.Engine.OM.UnitTests`. It comes there through `Microsoft.Build.Tasks.csproj` reference. Now `ManagedFileSystem` needs the `ChangeWaves` class, which is internal class of `Microsoft.Build.Framework`. Therefore, `Microsoft.Build.Engine.OM.UnitTests` wants access to internals of `Microsoft.Build.Framework`. 
The whole situation appeared because a Shared class needs  internals of `Microsoft.Build.Framework`.